clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -399,8 +410,42 @@ public void handleStoreDeleted(Store store) {
               + "may not be killed if admin helix messaging channel is disabled");
     }
 
-    // TODO: Wire configs into these params
-    KafkaValueSerializer kafkaValueSerializer = new OptimizedKafkaValueSerializer();
+    /**
+     * Register a callback function to handle the case when a new KME value schema is encountered when the server
+     * consumes messages from Kafka.
+     */
+    BiConsumer<Integer, Schema> newSchemaEncountered = (schemaId, schema) -> {
+      LOGGER.info("Encountered a new KME value schema (id = {}), proceed to register", schemaId);
+      try {
+        ControllerClientBackedSystemSchemaInitializer schemaInitializer =
```

Comment:
So. this logic will be triggered when encountering an unknown schema, and I see some minor issue:
 The resource is not reclaimed after each usage. Since this logic will create a new instance of `ControllerClientBackedSystemSchemaInitializer`, and it never be reused, so shall we close it after the call?

Reply:
Thanks for the review Gaojie.

That's correct. 
> So. this logic will be triggered when encountering an unknown schema.

Let me understand your concern better, `schemaInitializer` is a local variable and its memory would be good for GC, once it is out of the scope, right?

> The resource is not reclaimed after each usage. Since this logic will create a new instance of ControllerClientBackedSystemSchemaInitializer, and it never be reused, so shall we close it after the call?

Reply:
The `schemaInitializer` instance will allocate a ControllerClient to register system schema, so if we don't want to reuse this instance, shall we close the Controller client created internally?

Reply:
Oh, I see, the internal `controllerClient`. right, we should close it.